### PR TITLE
feat: add helper to identify computed values

### LINKED ISSUE #19: fix: [Vue warn]: Write operation failed: computed value is readonly.
**🐛 The bug**
At the moment, if a component that calls `useFetch` has a computed property returned from setup, it will give the following error:
```
[Vue warn]: Computed property was assigned to but it has no setter.
```

It still works correctly, but this is unattractive, and ideally we wouldn't be attempting to assign a value to a computed property at all.

### PR DESCRIPTION
**Pull request by mathe42**
fixes #19

### CODE REVIEW

**Comment by danielroe** [comment]
This doesn't appear to work for me. Could you perhaps use it in a test, so I can see how it's meant to work?

It seems that computed properties, once on the instance, have a getter and setter - but the setter just yields a warning - see [source](https://github.com/vuejs/vue/blob/33e669b22f69a1f9c9147528360fe0bba85534f0/src/core/instance/state.js#L233).

**Comment by mathe42** [comment]
When I start `node .\test\start-fixture.js` and go to http://localhost:3000/ssr-ref i get no errors in the console.

**Comment by mathe42** [comment]
And when I add 

`console.log(isComputed(computedVal), isComputed(''), isComputed(5), isComputed({}), isComputed(refValue), isComputed(ref({})))`

it logs true false false false false false as expected...

**Comment by mathe42** [comment]
:O

I use __lookupSetter__ wich is deprecated: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/__lookupSetter__

**Comment by mathe42** [comment]
Now it should work for you.

**Comment by danielroe** [comment]
Ah, I see. I raised the original issue because `useFetch` stringifies the entire component state - computed properties included. Unfortunately, once they are attached to an instance, they *do* have a setter, which is why we get the warning when viewing `/` in the fixture. It's not quite a bug because nothing happens in production, but it's unfortunate DX.

Ideally we would only set the property if `!isComputed(vm[key])`. But at the moment, I get `true` for both `ref` and `computed` properties of a `vm` - so even with this helper, it doesn't look like we're able to distinguish between them.

**Comment by mathe42** [comment]
I will add some tests maybe it is some strange difference in the browsers. For me (chromium based) it works...

File: src/fetch.ts [hunk 1]
```diff
@@ -161,12 +162,8 @@ export const useFetch = (callback: Fetch) => {
   onBeforeMount(() => {
     // Merge data
     for (const key in data) {
-      try {
+      if (isComputed((vm as any)[key])) {
```
Comment (danielroe) [inline src/fetch.ts:165]:
```suggestion
      if (!isComputed((vm as any)[key])) {
```

Comment (danielroe) [reply src/fetch.ts:165]: It would probably be better also to run this test on server side for a small performance win.

**Comment by mathe42** [comment]
@danielroe I fixed it :D

**Comment by danielroe** [comment]
@mathe42 This still gives the console warning when loading `/` in dev mode in Firefox and Chrome. The helper works to identify computed properties only before they are attached to a component instance.

But in order to be used in `useFetch`, we need to identify whether properties are computed *after* they have already been attached.

**Comment by mathe42** [comment]
Solution:

1. Merge #32 (not changes in fetch.ts).
2. On serverside wrap setup in the defineComponent function.
3. In wrapper check for each returned key - value pair if it contains a computed and add save the keys in a variable (maybe in the options)
4. On change tracking: ignore computed keys.

I'm not 100% shure how useFetch works but I think you (@danielroe ) find a way to handle it :D

**Comment by mathe42** [comment]
@danielroe This now should work properly.

Use the custom `createComponent` and computed is not hydrated.

**Comment by mathe42** [comment]
@danielroe can you review this?

**Comment by sseeland** [comment]
Why has this been closed? I still get this error and would really like to see this merged. The issue that this PR relates to is still open and is tagged "help wanted". Are there any problems with this PR, besides the missing review? I'm not really complaining, this is open source and volunteer work and I totally understand if people don't have the time for a review. I'm just curious because it looks to me that this is the solution to an issue that "just" needs to be merged.

**Comment by mathe42** [comment]
2 reasons:

1. The iscomputed helper is a hack.
2. UseFetch got a big refactor so i have to merge it. - I dont understand the useFetch method so I basicly dont know where to start.


In vue3 you can detect easy if a ref is a computedref but in @vue/Composition-API not.

If the vue Team adds a iscomputed helper i will check it out... I will create a Feature Request for that...

**Comment by sseeland** [comment]
OK, got it. Thanks for the explanation.

**Comment by mathe42** [comment]
See https://github.com/vuejs/composition-api/issues/445

**Pull request closed without merging**